lib/linter/apply-disable-directives.js

```diff
@@ -210,11 +209,14 @@ function applyDirectives(options) {
     const usedDisableDirectives = new Set();
 
     for (const problem of options.problems) {
+        let nextDirectiveIndex = 0;
+
         while (
             nextDirectiveIndex < options.directives.length &&
             compareLocations(options.directives[nextDirectiveIndex], problem) <= 0
         ) {
             const directive = options.directives[nextDirectiveIndex++];
+            const suppression = { kind: "directive", justification: directive.unprocessedDirective.justification };
```

Comment:
Would it make sense to add location information to the suppression to show where the directive is?

Reply:
It's a good idea. The location info can be leveraged for auditing purposes as well. I also find a [location](https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html#_Toc10127928) property in `suppressions` in the SARIF format.

But I'd like to add location info to the suppressions in another PR because:
1. The type `location` in `suppressions` is needed to be discussed. Shall we use the `line` and the `column` in `unprocessedDirective` directly or we use the similar structure in `LintMessage` (`line`/`column`/`endLine`/`endColumn`)?
2. This PR is huge enough due to a lot of modification in core logic and tests.